### PR TITLE
Fixed #25441 -- Added support for negative filesize to filesizeformat template filter.

### PR DESCRIPTION
**Pull request by darkryder**
See #5327

### CODE REVIEW

File: django/template/defaultfilters.py [hunk 1]
```diff
@@ -871,7 +875,9 @@ def filesizeformat(bytes):
     else:
         value = ugettext("%s PB") % filesize_number_format(bytes / PB)
 
-    return avoid_wrapping(value)
+    if negative:
+        value = "&minus;%s" % value
```
Comment (charettes) [inline django/template/defaultfilters.py]: Is there a reason you are using`&minus;` instead of `-`? It won't work in non-HTML templates.

Comment (darkryder) [inline django/template/defaultfilters.py]: Oh, I didn't think of that, I'll change it.

File: django/template/defaultfilters.py [hunk 2]
```diff
@@ -858,6 +858,10 @@ def filesizeformat(bytes):
     TB = 1 << 40
     PB = 1 << 50
 
+    negative = True if bytes < 0 else False
```
Comment (claudep) [inline django/template/defaultfilters.py]: `negative = bytes < 0`

**Comment by claudep** [comment]
Even if it's unrelated to the patch, it's unfortunate that we use a native type name like `bytes` as a variable name. Maybe `bytes_`? (could be a separate commit/PR, though).

**Comment by darkryder** [comment]
This messes with the line length though, and I don't want to break lines, as in this case it might break the readability flow.

**Comment by timgraham** [comment]
Merged in ce7dd1273e5ee3d927dbfda0b38aff2517fe004b and dc4c8df7aac81b5d68e40c1327fb362790cde70d, thanks.

**Pull request closed without merging**